taichi/transforms/type_check.cpp

```diff
@@ -196,9 +196,17 @@ class TypeCheck : public IRVisitor {
   }
 
   void visit(UnaryOpStmt *stmt) override {
-    stmt->ret_type = stmt->operand->ret_type;
+    auto operand_type = stmt->operand->ret_type;
```

Comment:
Are these changes related to this PR?

Reply:
Yup, we do have some test cases with `real_matrix_scalarize=False` such as `test_matrix_return_real_matrix`. These tests used to work because of the scalarization at Python level, so after removing the `expand_exprs()` we'll have to fix the issues with CodeGen and runtime.